File: library/src/main/java/com/pengrad/telegrambot/model/request/InputMediaAnimation.java
package com.pengrad.telegrambot.model.request;

import com.pengrad.telegrambot.request.ContentTypes;

import java.io.File;
import java.io.Serializable;

/**
 * Stas Parshin
 * 28 July 2018
 */
public class InputMediaAnimation extends InputMedia<InputMediaAnimation> implements Serializable {
    private final static long serialVersionUID = 0L;

    private Integer width, height, duration;

    public InputMediaAnimation(String media) {
        super("animation", media);
    }

    public InputMediaAnimation(File media) {
        super("animation", media);
    }

    public InputMediaAnimation(byte[] media) {
        super("animation", media);
    }

    public InputMediaAnimation width(Integer width) {
        this.width = width;
        return this;
    }

    public InputMediaAnimation height(Integer height) {
        this.height = height;
        return this;
    }

    public InputMediaAnimation duration(Integer duration) {
        this.duration = duration;
        return this;
    }

    @Override
    public String getDefaultFileName() {
        return ContentTypes.GIF_FILE_NAME;
    }

    @Override
    public String getContentType() {
        return ContentTypes.GIF_MIME_TYPE;
    }
}

File: core/src/main/kotlin/com/pengrad/telegrambot/model/request/InputMediaAnimation.kt
package com.pengrad.telegrambot.model.request

import com.pengrad.telegrambot.request.ContentTypes

import java.io.File
import java.io.Serializable

/**
 * Stas Parshin
 * 28 July 2018
 */
class InputMediaAnimation : InputMedia<InputMediaAnimation>, Serializable {

    private var width: Int? = null
    private var height: Int? = null
    private var duration: Int? = null

    override val defaultFileName: String
        get() = ContentTypes.GIF_FILE_NAME

    override val contentType: String
        get() = ContentTypes.GIF_MIME_TYPE

    constructor(media: String) : super("animation", media) {}

    constructor(media: File) : super("animation", media) {}

    constructor(media: ByteArray) : super("animation", media) {}

    fun width(width: Int?): InputMediaAnimation {
        this.width = width
        return this
    }

    fun height(height: Int?): InputMediaAnimation {
        this.height = height
        return this
    }

    fun duration(duration: Int?): InputMediaAnimation {
        this.duration = duration
        return this
    }

    companion object {
        private const val serialVersionUID = 0L
    }
}
